### PR TITLE
Fixes #1105 Double clicking on spatial structure with attached projec…

### LINKED ISSUE #1105: Double clicking on spatial structure with attached project= > error
Feels like it's related to #1084  fix from this morning @rwmcintosh but maybe not
Application:
MoBi® 12.0.0

Object reference not set to an instance of an object.

Stack trace:
```
at MoBi.Presentation.Presenter.HierarchicalStructurePresenter.createNeighborhoodId(IWithId me, IWithId myNeighbor) in C:\dev\MoBi\src\MoBi.Presentation\Presenter\HierarchicalStructurePresenter.cs:line 91
   at MoBi.Presentation.Presenter.HierarchicalStructurePresenter.<neighborsOf>d__7.MoveNext() in C:\dev\MoBi\src\MoBi.Presentation\Presenter\HierarchicalStructurePresenter.cs:line 77
   at System.Linq.Enumerable.<UnionIterator>d__67`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at MoBi.Presentation.Presenter.HierarchicalStructurePresenter.GetChildrenSorted(IContainer container, Func`2 predicate) in C:\dev\MoBi\src\MoBi.Presentation\Presenter\HierarchicalStructurePresenter.cs:line 67
   at MoBi.Presentation.Presenter.HierarchicalStructurePresenter.GetChildObjects(ObjectBaseDTO dto, Func`2 predicate) in C:\dev\MoBi\src\MoBi.Presentation\Presenter\HierarchicalStructurePresenter.cs:line 51
   at MoBi.UI.Views.HierarchicalStructureView.<Show>b__4_0(ObjectBaseDTO dto) in C:\dev\MoBi\src\MoBi.UI\Views\HierarchicalStructureView.cs:line 59
   at MoBi.Presentation.Mappers.ObjectBaseDTOToSpatialStructureNodeMapper.<MapFrom>b__1_0(ObjectBaseDTO x) in C:\dev\MoBi\src\MoBi.Presentation\Mappers\ObjectBaseDTOToSpatialStructureNodeMapper.cs:line 27
   at MoBi.Presentation.Nodes.HierarchicalStructureNode.get_Children() in C:\dev\MoBi\src\MoBi.Presentation\Nodes\HierarchicalStructureNode.cs:line 26
   at OSPSuite.UI.Controls.UxTreeView.<>c__DisplayClass42_0.<AddNode>b__0()
   at OSPSuite.UI.Controls.UxTreeList.DoWithinBatchUpdate(Action action)
   at OSPSuite.UI.Controls.UxTreeView.AddNode(ITreeNode nodeToAdd)
   at OSPSuite.Utility.Extensions.EnumerableExtensions.Each[T](IEnumerable`1 list, Action`1 action)
   at OSPSuite.UI.Controls.UxTreeView.<>c__DisplayClass42_0.<AddNode>b__0()
   at OSPSuite.UI.Controls.UxTreeList.DoWithinBatchUpdate(Action action)
   at OSPSuite.UI.Controls.UxTreeView.AddNode(ITreeNode nodeToAdd)
   at MoBi.UI.Views.HierarchicalStructureView.AddRoot(ObjectBaseDTO dto) in C:\dev\MoBi\src\MoBi.UI\Views\HierarchicalStructureView.cs:line 96
   at OSPSuite.Utility.Extensions.EnumerableExtensions.Each[T](IEnumerable`1 list, Action`1 action)
   at MoBi.UI.Views.HierarchicalStructureView.Show(IEnumerable`1 roots) in C:\dev\MoBi\src\MoBi.UI\Views\HierarchicalStructureView.cs:line 60
   at MoBi.Presentation.Presenter.HierarchicalSpatialStructurePresenter.InitializeWith(SpatialStructure spatialStructure) in C:\dev\MoBi\src\MoBi.Presentation\Presenter\HierarchicalSpatialStructurePresenter.cs:line 61
   at MoBi.Presentation.Presenter.HierarchicalSpatialStructurePresenter.Edit(SpatialStructure spatialStructure) in C:\dev\MoBi\src\MoBi.Presentation\Presenter\HierarchicalSpatialStructurePresenter.cs:line 66
   at MoBi.Presentation.Presenter.EditSpatialStructurePresenter.Edit(MoBiSpatialStructure spatialStructure) in C:\dev\MoBi\src\MoBi.Presentation\Presenter\EditSpatialStructurePresenter.cs:line 66
   at MoBi.Presentation.Presenter.EditBuildingBlockPresenterBase`4.Edit(Object objectToEdit) in C:\dev\MoBi\src\MoBi.Presentation\Presenter\EditBuildingBlockPresenterBase.cs:line 41
   at MoBi.Presentation.Tasks.Edit.EditTasksForBuildingBlock`1.editEntity(T entity) in C:\dev\MoBi\src\MoBi.Presentation\Tasks\Edit\EditTasksForBuildingBlock.cs:line 56
   at MoBi.Presentation.Tasks.Edit.EditTasksForBuildingBlock`1.Edit(T buildingBlock) in C:\dev\MoBi\src\MoBi.Presentation\Tasks\Edit\EditTasksForBuildingBlock.cs:line 45
   at MoBi.Presentation.UICommand.EditCommandFor`1.PerformExecute() in C:\dev\MoBi\src\MoBi.Presentation\UICommand\EditCommandFor.cs:line 18
   at OSPSuite.Presentation.UICommands.ObjectUICommand`1.Execute()
   at OSPSuite.Presentation.UICommands.UICommandExtensions.ExecuteWithinExceptionHandler(IUICommand uiCommand, IEventPublisher eventPublisher, IChangePropagator changePropagator)
```

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

**Comment by rwmcintosh** [comment]
This is probably it, but if you can attach the project, I'll verify it.

**Comment by rwmcintosh** [comment]
Project from https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/2102

File: src/MoBi.Presentation/Presenter/HierarchicalStructurePresenter.cs [hunk 1]
```diff
@@ -88,7 +88,7 @@ private IEnumerable<ObjectBaseDTO> neighborsOf(NeighborhoodBuilder neighborhoodB
       /// <returns>An Id that combines the two Ids of the neighborhood</returns>
       private string createNeighborhoodId(IWithId me, IWithId myNeighbor)
       {
-         return $"{me.Id}-{myNeighbor.Id}";
+         return $"{me?.Id}-{myNeighbor?.Id}";
```
Comment (msevestre) [inline src/MoBi.Presentation/Presenter/HierarchicalStructurePresenter.cs:91]: I am not quite getting it. Why aren't we using the id of the neighborhood? 
Also two neighborhood with the same source but no target would have the same id right?

**Review by msevestre — CHANGES_REQUESTED** [review]
I don't understand why we are generating id like this

**Review by msevestre — APPROVED** [review]

**Pull request merged**